test/httpPipelineTests.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import * as assert from "assert";
+import { HttpClient } from "../lib/httpClient";
+import { HttpPipeline } from "../lib/httpPipeline";
+import { HttpRequest } from "../lib/httpRequest";
+import { HttpResponse } from "../lib/httpResponse";
+import { userAgentRequestPolicyFactory } from "../lib/policies/userAgentRequestPolicyFactory";
+import { BaseRequestPolicy } from "../lib/requestPolicy";
+import { InMemoryHttpResponse } from "./inMemoryHttpResponse";
+
+describe("HttpPipeline", () => {
+    it("should send requests when no request policies are assigned", () => {
+        const httpClient: HttpClient = (request: HttpRequest) => {
+            return Promise.resolve(new InMemoryHttpResponse(request, 200, {}, "hello"));
+        };
+
+        const httpPipeline = new HttpPipeline([], { httpClient: httpClient });
+
+        const httpRequest = new HttpRequest("GET", "http://www.example.com", {});
+        return httpPipeline.send(httpRequest)
+            .then((response: HttpResponse) => {
```

Comment:
Let's make this an `await` and add the `async` modifier to the test lambda on line 13

Reply:
same for every .then() in this changeset really

Reply:
I'll change it to async/await.